src/core/algorithms/algo_factory.h

```diff
@@ -71,12 +72,28 @@ template <typename OptionMap>
 void LoadAlgorithm(Algorithm& algorithm, OptionMap&& options) {
     ConfigureFromFunction(algorithm, [&options](std::string_view option_name) {
         using namespace config::names;
+        namespace fs = std::filesystem;
         if (option_name == kTable && options.find(std::string{kTable}) == options.end()) {
             config::InputTable parser = std::make_shared<CSVParser>(
-                    details::ExtractOptionValue<std::filesystem::path>(options, kCsvPath),
+                    details::ExtractOptionValue<fs::path>(options, kCsvPath),
                     details::ExtractOptionValue<char>(options, kSeparator),
                     details::ExtractOptionValue<bool>(options, kHasHeader));
             return boost::any{parser};
+        } else if (option_name == kTables && options.find(std::string{kTables}) == options.end()) {
```

Comment:
This case handling looks bad, we certainly don't want here a big if-else spaghetti handling all cases where input data option may be constructed from several others. Let's leave it as it is for now, and I'll create an issue